common-app-lib/src/cmr/common_app/services/kms_fetcher.clj

```diff
@@ -51,66 +52,68 @@
   "The key used to store the KMS cache in the system cache map."
   :kms)
 
+(def kms-cache-ttl
+  "The Time-To-Live value, use nil to never expire."
+  nil)
+
+;; called by system.clj across several applications
 (defn create-kms-cache
   "Used to create the cache that will be used for caching KMS keywords. All applications caching
   KMS keywords should use the same fallback cache to ensure functionality even if GCMD KMS becomes
   unavailable."
   []
   (redis-cache/create-redis-cache {:keys-to-track [kms-cache-key]
                                    :read-connection (redis-config/redis-read-conn-opts)
-                                   :primary-connection (redis-config/redis-conn-opts)}))
+                                   :primary-connection (redis-config/redis-conn-opts)
+                                   :ttl kms-cache-ttl}))
 
+;; Called indirectly by bootstrap. This may call KMS service directly through transmit.
```

Comment:
I added a few comments like this, and I would like to keep them because they all give hints on function usage. This I feel is important to document here because we have very frequently changed the KMS cache code. This code has huge impacts and I would like us to try to not spread usage of functions out more then we need to as the code is hard enough to understand is it stands.